packages/types/src/pricing/service.ts

```diff
@@ -69,8 +72,23 @@ export interface IPricingModuleService {
     sharedContext?: Context
   ): Promise<PriceSetDTO[]>
 
+  removeRules(
+    data: RemovePriceSetRulesDTO[],
+    sharedContext?: Context
+  ): Promise<void>
+
   delete(ids: string[], sharedContext?: Context): Promise<void>
 
+  addPrices(
```

Comment:
todo: when passing one item we don't necessarely expect to receive an array as a result, why don't we do like in the core? if the user pass an item we return an item if he passes multiple item then we return an array of result? also, for the interfaces it might be better to express that as a proper overload wdyt?

```ts
addPrices(
    data: AddPricesDTO,
    sharedContext?: Context
  ): Promise<PriceSetDTO>

addPrices(
    data: AddPricesDTO[],
    sharedContext?: Context
  ): Promise<PriceSetDTO[]>
```

Reply:
same goes for all methods

Reply:
good plan 👍 let's leave a large refactor of the entire service with this pattern to a follow up pr to not hold this up and clutter it